src/uct/ib/base/ib_md.c

```diff
@@ -1003,7 +1013,12 @@ uct_ib_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
             ucs_trace("created atomic key 0x%x for 0x%x", memh->atomic_rkey,
                       memh->lkey);
         } else if (status == UCS_ERR_UNSUPPORTED) {
-            /* ignore for atomic MR */
+            if (flags & UCT_MD_MKEY_PACK_FLAG_INVALIDATE) {
+                ucs_assert(md->cap_flags & UCT_MD_FLAG_INVALIDATE_AMO);
```

Comment:
what is this assert checking?
could it be done in an earlier stage? why only assert when we fail?